sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SpecificInternalRow.scala

```diff
@@ -208,10 +208,9 @@ final class SpecificInternalRow(val values: Array[MutableValue]) extends BaseGen
   def this(dataTypes: Seq[DataType]) = {
     // SPARK-32550: use while loop instead of map
     this(new Array[MutableValue](dataTypes.length))
-    val length = values.length
     var i = 0
-    while (i < length) {
-      values(i) = dataTypeToMutableValue(dataTypes(i))
+    dataTypes.foreach { dt =>
```

Comment:
Let's also add a short comment. Using plan while is actually recommanded per the guide at https://github.com/databricks/scala-style-guide#traversal-and-zipwithindex so it's likely people attempt to use while here in the future

Reply:
Heh we should probably update the style guide to note exactly this - that is only an 'optimization' if you have constant-time access to both sequences.

Reply:
Sorry, I fell asleep yesterday.......  and Address [9976bf5](https://github.com/apache/spark/pull/29529/commits/9976bf55a6ec7833e31b03433cff09acc696c644) update the comments.